helix-core/src/main/java/org/apache/helix/util/InstanceValidationUtil.java

```diff
@@ -244,13 +248,24 @@ public static boolean hasErrorPartitions(HelixDataAccessor dataAccessor, String
               continue;
             }
 
+            // If the state is init state, we add appropriate messages
+            if (stateMap.get(siblingInstance).equals(stateModelDefinition.getInitialState())) {
+              if (!unhealthyPartitions.containsKey(partition)) {
+                unhealthyPartitions.put(partition, new ArrayList<>());
+              }
+              unhealthyPartitions.get(partition).add(PARTITION_INITIAL_STATE_FAIL);
```

Comment:
NIT: we can combine it to:

unhealthyPartitions.computeIfAbsent(partition, list -> new ArrayList<>()).add(PARTITION_INITIAL_STATE_FAIL);

Same logic can apply to following piece.

Reply:
Done.